pkg/system/xattrs_linux.go

```diff
@@ -10,24 +10,23 @@ func Lgetxattr(path string, attr string) ([]byte, error) {
 	dest := make([]byte, 128)
 	sz, errno := unix.Lgetxattr(path, attr, dest)
 
-	switch {
-	case errno == unix.ENODATA:
-		return nil, nil
-	case errno == unix.ERANGE:
-		// 128 byte array might just not be good enough. A dummy buffer is used
-		// to get the real size of the xattrs on disk
+	for errno == unix.ERANGE {
```

Comment:
I don't see one directly, but is there a potential situation where this ends up in an infinite loop? And/or does this need throttling?

Reply:
There's a very slight chance that we'll ever get an `ERANGE` for the second time, and if we'll get it more than that it means the kernel is severely broken, and the world is coming to the end...

Reply:
Throttling is not needed here either, this is not supposed to fail with ERANGE repeatedly. 

Here's the flow:

1. we get the attributes using the default buffer size of 128 bytes.
2. if the actual size is larger (ERANGE), we
2.1 get the actual size
2.2 get the attributes using the buffer of the actual size

Now, in an _extremely_ rare scenario that the xattrs changes between 2.1 and 2.2, and the new ones are larger, we need to repeat 2 one more time.

Reply:
❤️ Thanks for your in-depth answer! (better to ask, than to overlook 🤗). From your answer, I agree that the corner cases would be _very_ unlikely, so not something to care about 👍